windows/src/test/test-kmcomapi-win64/Keyman.Test.Console.KeymanAPIHost.pas

```diff
@@ -279,157 +379,284 @@ procedure TKeymanAPITest.Test_IKeymanHotkeys;
-  // test IKeymanKeyboardFile and IKeymanKeyboardInstalled
+  // Tested in IKeymanKeyboardFile and IKeymanKeyboardInstalled
+  Assert.Pass;
 end;
 
 procedure TKeymanAPITest.Test_IKeymanKeyboardFile;
 var
-  f: IKeymanKeyboardFile;
-begin
-  f := k.Keyboards.GetKeyboardFromFile(FTestPath + 'test.kmx');
-  Assert.AreEqual('กขฃ', f.GetCharsUsed);
-  Assert.AreEqual('(C) 2017 SIL International', f.Copyright);
-  Assert.AreEqual('eng', f.DefaultBCP47Languages);
-  Assert.AreEqual(1036, f.DefaultPrimaryLanguage);
-  Assert.AreEqual('x040C x040C', f.DefaultWindowsLanguages);
-//  Assert.IsNotNull(f.DefaultHotkey);  TODO: Resolve DefaultHotkey implementation -- remove from reg, read from kmx in --installed, read from kmx in --file as well
-//  Assert.AreEqual(False, f.DefaultHotkey.IsEmpty);
-//  Assert.AreEqual(HK_ALT or HK_CTRL, f.DefaultHotkey.Modifiers);
-//  Assert.AreEqual(Ord('T'), f.DefaultHotkey.VirtualKey);
-//  Assert.AreEqual(0, f.DefaultHotkey.RawValue);
-//  Assert.AreEqual(0, f.DefaultHotkey.Target);
-  Assert.AreEqual(keUnicode, f.Encodings);
-  Assert.AreEqual(FTestPath + 'test.kmx', f.Filename);
-  Assert.AreEqual('test', f.ID);
-  Assert.AreEqual(kltPositional, f.LayoutType);
-  Assert.AreEqual('Test Message', f.Message);
-  Assert.AreEqual('Test Keyboard', f.Name);
-  Assert.AreEqual('1.2.3', f.Version);
-  Assert.IsNotNull(f.Bitmap);
-//  Assert.AreEqual(f.SerializeXML(...), False);
-end;
-
-procedure TKeymanAPITest.Test_IKeymanKeyboardInstalled;
+  kbd: IKeymanKeyboardFile;
 begin
-  Assert.Fail();
+  kbd := k.Keyboards.GetKeyboardFromFile(FTestPath + 'test.kmx');
+  CheckKeyboardProperties(kbd);
+  // kbd.Install()... Todo: test installation (forced installation is tested in IKeymanKeyboardInstalled)
 end;
 
 procedure TKeymanAPITest.Test_IKeymanKeyboardLanguageInstalled;
 begin
-  Assert.Fail();
+  // Tested in Test_IKeymanKeyboardInstalled
+  Assert.Pass;
 end;
 
 procedure TKeymanAPITest.Test_IKeymanKeyboardLanguagesInstalled;
 begin
-  Assert.Fail();
+  // Tested in Test_IKeymanKeyboardInstalled
+  Assert.Pass;
 end;
 
 procedure TKeymanAPITest.Test_IKeymanKeyboardOption;
 begin
-  Assert.Fail();
+  //TODO
 end;
 
 procedure TKeymanAPITest.Test_IKeymanKeyboardOptions;
 begin
-  Assert.Fail();
+  //TODO
 end;
 
 procedure TKeymanAPITest.Test_IKeymanKeyboards;
 begin
-  Assert.Fail();
+  // Tested in Test_IKeymanKeyboardInstalled
+  Assert.Pass;
 end;
 
 procedure TKeymanAPITest.Test_IKeymanKeyboardsInstalled;
 begin
-  Assert.Fail();
+  // Tested in Test_IKeymanKeyboardInstalled
+  Assert.Pass;
 end;
 
+//
+// IKeymanLanguage*
+//
+
 procedure TKeymanAPITest.Test_IKeymanLanguage;
 begin
-  Assert.Fail();
+  Assert.IsTrue(k.Languages.Count > 0);
+
+  k.Languages[0].BCP47Code;
+  k.Languages[0].HKL;
+  k.Languages[0].Hotkey;
+  k.Languages[0].KeymanKeyboardLanguage;
+  k.Languages[0].LangID;
+  k.Languages[0].LayoutName;
+  k.Languages[0].LocaleName;
+  k.Languages[0].ProfileGUID;
+  k.Languages[0].ClassID;
```

Comment:
These are testing that the property accessors work correctly?

Reply:
Yes. It'd be nice to do more one day. But at least we are getting some coverage with these!